lib/codeowners/cli/main.rb

```diff
@@ -129,16 +169,19 @@ def apply_suggestion(line, suggestion)
           pattern_change(line)
         when 'd'
           line.remove!
+        when 'q'
+          @quit = true
```

Comment:
With `@quit` the loop in `checker` will continue to run just all the calls will be aborting early, maybe instead we should use an exception to handle the flow?

```ruby
throw :user_quit
```
and in https://github.com/toptal/codeowners-checker/blob/master/lib/codeowners/checker.rb#L69
we can add:
```ruby
  def useless_pattern
    catch(:user_quit) do # <- added line
      codeowners.select do |line|
        next unless line.pattern?

        unless pattern_has_files(line.pattern)
          @when_useless_pattern&.call(line)
          true
        end
      end
    end  # <- added line
  end
```

Reply:
I put it to `check!` method so we really quit the program, otherwise we interrupt only one of the checks.